src/ReactLeafletMap/layers/VectorLayer/index.js

```diff
@@ -25,31 +25,38 @@ class VectorLayer extends React.PureComponent {
     }
 
     getDefaultStyleObject(feature) {
-        return mapStyle.getStyleObject(feature.properties, this.props.style);
+        return mapStyle.getStyleObject(feature.properties, this.props.style || constants.vectorFeatureStyle.defaultFull);
     }
 
-    getAccentedStyleObject(styleDefinition, fallbackStyleDefinition) {
-        const style = {"rules":[{"styles": [styleDefinition || fallbackStyleDefinition]}]};
-        return mapStyle.getStyleObject(null, style, true);
-    }
 
     getFeatureDefaultStyle(feature, defaultStyleObject) {
         return this.getFeatureLeafletStyle(feature, defaultStyleObject);
     }
 
-    getFeatureAccentedStyle(feature, defaultStyleObject, accentedStyleObject) {
-        const style = {...defaultStyleObject, ...accentedStyleObject};
+    getFeatureAccentedStyle(feature, defaultStyleObject, styleDefinition) {
+        const style = {...defaultStyleObject, ...styleDefinition};
         return this.getFeatureLeafletStyle(feature, style);
     }
 
     getFeatureLeafletStyle(feature, style) {
-        let finalStyle = {
-            color: style.outlineColor,
-            weight: style.outlineWidth,
-            opacity: style.outlineOpacity,
-            fillColor: style.fill,
-            fillOpacity: style.fillOpacity
-        };
+        let finalStyle = {};
+
+        finalStyle.color = style.outlineColor ? style.outlineColor : null;
+        finalStyle.weight = style.outlineWidth ? style.outlineWidth : 0;
+        finalStyle.opacity = style.outlineOpacity ? style.outlineOpacity : 1;
+        finalStyle.fillOpacity = style.fillOpacity ? style.fillOpacity : 1;
+        finalStyle.fillColor = style.fill;
+
+        if (!style.fill) {
+            finalStyle.fillColor = null;
+            finalStyle.fillOpacity = 0;
+        }
+
+        if (!style.outlineColor || !style.outlineWidth) {
+            finalStyle.color = null;
+            finalStyle.opacity = 0;
+            finalStyle.weight = 0;
+        }
```

Comment:
Are these general defaults, or framework-specific?

Reply:
Framework-specific, necessary to draw (or rather not to draw) desired properties and reset framework defaults